packages/plugin/src/controllers/api/fields/GroupsController.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Solspace\Freeform\controllers\api\fields;
+
+use Solspace\Freeform\Bundles\Fields\Types\FieldTypesProvider;
+use Solspace\Freeform\controllers\BaseApiController;
+use Solspace\Freeform\Freeform;
+use Solspace\Freeform\Records\FieldTypeGroupRecord;
+
+class GroupsController extends BaseApiController
+{
+    private FieldTypesProvider $fieldTypesProvider;
+
+    public function __construct($id, $module, $config = [], FieldTypesProvider $fieldTypesProvider)
+    {
+        parent::__construct($id, $module, $config);
+        $this->fieldTypesProvider = $fieldTypesProvider;
+    }
+
+    protected function get(): object
+    {
+        $freeform = Freeform::getInstance();
+        $groups = FieldTypeGroupRecord::find()->all();
+        $types = $this->fieldTypesProvider->getRegisteredTypes();
+
+        $response = (object) [
+            'types' => [],
+            'groups' => [
+                'hidden' => [],
+                'grouped' => [],
+            ],
+        ];
+
+        if ($freeform->isPro()) {
+            $hiddenFieldTypes = $freeform->settings->getSettingsModel()->hiddenFieldTypes;
+            $unassignedTypes = array_diff($types, array_merge(...array_column($groups, 'types')), $hiddenFieldTypes);
+
+            $response->types = [...$unassignedTypes];
+            $response->groups = (object) [
+                'hidden' => $hiddenFieldTypes,
+                'grouped' => $groups,
+            ];
+
+            return $response;
+        }
+
+        $filteredGroups = array_map(function ($group) use ($types) {
+            $filteredTypes = array_filter($group->types, function ($type) use ($types) {
+                return \in_array($type, $types);
+            });
```

Comment:
There's a nice shorthand for writing anonymous function in php now.

```suggestion
            $filteredTypes = array_filter(
                $group->types, 
                fn ($type) => \in_array($type, $types)
            );
```